pkg/k8shandler/status.go

```diff
@@ -47,15 +45,55 @@ func updateNodeStatus(node *nodeState, dpl *v1alpha1.ElasticsearchStatus) {
 	if node.Actual.StatefulSet != nil {
 		nodeStatus.StatefulSetName = node.Actual.StatefulSet.Name
 	}
+
+	if node.Desired.Roles != nil {
+		nodeStatus.Roles = node.Desired.Roles
+	}
 	dpl.Nodes = append(dpl.Nodes, nodeStatus)
 }
 
-func AddStatusMessage(message string, dpl *v1alpha1.Elasticsearch) error {
-	dpl.Status.Message = message
-	err := sdk.Update(dpl)
+func clusterHealth(dpl *v1alpha1.Elasticsearch) string {
+	pods, err := listPods(dpl.Name, dpl.Namespace)
 	if err != nil {
-		return fmt.Errorf("failed to update Elasticsearch status: %v", err)
+		return healthUnknown
 	}
 
-	return nil
+	// no elasticsearch pods were found
+	if len(pods.Items) == 0 {
+		return healthUnknown
+	}
+
+	// use arbitrary pod
+	pod := pods.Items[0]
```

Comment:
Issue using an arbitrary pod is what if this pod is not accepting connections (i.e. its starting)?  This will error though the cluster could be functional

Reply:
What if we use the service end point instead? Or we could look for pods that are `ready` only

Reply:
There has been discussion about removing `.phase` from `.pod.status`[1]

[1] https://github.com/kubernetes/kubernetes/issues/7856